firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigFetchHandler.java

```diff
@@ -156,13 +163,67 @@ public Task<FetchResponse> fetch() {
    *     updates, the {@link FetchResponse}'s configs will be {@code null}.
    */
   public Task<FetchResponse> fetch(long minimumFetchIntervalInSeconds) {
+
+    // Make a copy to prevent any concurrency issues between Fetches.
+    Map<String, String> copyOfCustomHttpHeaders = new HashMap<>(customHttpHeaders);
+    copyOfCustomHttpHeaders.put(X_FIREBASE_RC_FETCH_TYPE, BASE_FETCH_TYPE + "/" + 1);
+
+    return fetchedConfigsCache
+        .get()
+        .continueWithTask(
+            executor,
+            (cachedFetchConfigsTask) ->
+                fetchIfCacheExpiredAndNotThrottled(
+                    cachedFetchConfigsTask,
+                    minimumFetchIntervalInSeconds,
+                    copyOfCustomHttpHeaders));
+  }
+
+  /**
+   * Starts fetching configs from the Firebase Remote Config server.
+   *
+   * <p>Guarantees consistency between memory and disk; fetched configs are saved to memory only
+   * after they have been written to disk.
+   *
+   * <p>Fetches even if the read of the fetch cache fails (assumes there are no cached fetched
+   * configs in that case).
+   *
+   * <p>If the fetch request could not be created or there was error connecting to the server, the
+   * returned Task throws a {@link FirebaseRemoteConfigClientException}.
+   *
+   * <p>If the server responds with an error, the returned Task throws a {@link
+   * FirebaseRemoteConfigServerException}.
+   *
+   * <p>If any of the following is true, then the returned Task throws a {@link
+   * FirebaseRemoteConfigFetchThrottledException}:
+   *
+   * <ul>
+   *   <li>The backoff duration from a previous throttled exception has not expired,
+   *   <li>The backend responded with a throttled error, or
+   *   <li>The backend responded with unavailable errors for the last two fetch requests.
+   * </ul>
+   *
+   * @return A {@link Task} representing a Realtime fetch call that returns a {@link FetchResponse}
+   *     with the configs fetched from the backend. If the backend was not called or the backend had
+   *     no updates, the {@link FetchResponse}'s configs will be {@code null}.
+   */
+  public Task<FetchResponse> realtimeFetch(
```

Comment:
It feels a bit off to me that half the header comes from calling a particular method (`realtimeFetch`) and half from a param passed the method (`fetchAttemptNumber`). It also breaks the encapsulation of fetch within realtime to have realtime implementation details here.

Since the type and attempt are fetch details and realtime is just making use of them, passing both in (i.e. `fetchWithMetadata(long minimumFetchIntervalInSeconds, FetchType fetchType, int fetchAttempt)`) could make sense. A `FetchType` enum could make sure the value passed in makes sense.

Reply:
I guess this makes sense. I updated it to create a new method that isn't so closely tied to Realtime.